drivers/volume/portworx/portworx.go

```diff
@@ -642,22 +641,40 @@ func (d *portworx) GetNodeForVolume(vol *torpedovolume.Volume) (*node.Node, erro
 		}
 	}
 
-	pxVol := v.(*api.Volume)
-	for _, n := range node.GetStorageDriverNodes() {
-		if n.VolDriverNodeID == pxVol.AttachedOn {
-			return &n, nil
+	r := func() (interface{}, bool, error) {
+		pxVol := v.(*api.Volume)
+		for _, n := range node.GetStorageDriverNodes() {
+			if n.VolDriverNodeID == pxVol.AttachedOn {
+				return &n, false, nil
+			}
+		}
+
+		// Snapshots may not be attached to a node
+		if pxVol.Source.Parent != "" {
+			logrus.Debugf("Volume with ID: %s is a snapshot", pxVol.Id)
+			return nil, false, nil
+		}
+
+		return nil, true, &ErrFailedToInspectVolume{
```

Comment:
need to define a new error called `ErrFailedToAttachVolume` might be confusing for new users if we print the failure to be for inspect.

Reply:
Made custom err for ErrFailedToValidateAttachment